testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,526 @@
+                            ob_0 = xstream.fromXML(state0);
+                        }
+                        try {
+                            if (AccessibleObject.class.isAssignableFrom(ob_0.getClass())) {
+                                ((AccessibleObject) ob_0).setAccessible(true);
+                            }
+                        } catch (NullPointerException npex) {
+                            npex.printStackTrace();
+                        }
+                        if (threadLocal) {
+                            Object tmp = fieldList[i].get(null);
+                            ((ThreadLocal)tmp).set(ob_0);
+                            ob_0 = tmp;
+                        }
+
+                        FieldUtils.writeField(fieldList[i], (Object)null, ob_0, true);
+
+                        String output = fieldName + " set\n";
+                        Files.write(Paths.get(reflectionFile), output.getBytes(),
+                                StandardOpenOption.APPEND);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                        String outputNormalError = fieldName + " reflectionError: " + e + "\n";
+                        Files.write(Paths.get(reflectionFile), outputNormalError.getBytes(),
+                                StandardOpenOption.APPEND);
+                    }
+                }
+            } catch (Exception e) { // *** check the kind of exception
+                e.printStackTrace();
+            }
+        } catch (Exception e) {
+            String output = fieldName + " deserializeError: " + e + "\n";
+            Files.write(Paths.get(reflectionFile), output.getBytes(),
+                    StandardOpenOption.APPEND);
+        }
+    }
+
+    @Override
+    public void capture() {
+        try {
+            String state = Configuration.config().getProperty("statecapture.state", "");
+            if (subxmlFold.isEmpty() || rootFile.isEmpty()) {
+                System.out.println("WARNING: The subxml folder or root folder are not provided, thus it will not do capturing.");
+                return;
+            }
+            if (!state.equals("eagerload")) {
+                capture_real();
+            } else {
+                capture_class();
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Adds the current serialized reachable state to the currentTestStates list
+     * and the current roots to the currentRoots list.
+     * @throws IOException
+     */
+    private void capture_real() throws IOException {
+        PrintWriter writer;
+
+        String subxmlDir = createSubxmlFold();
+
+        Set<String> allFieldName = new HashSet<String>();
+        Class[] loadedClasses = MainAgent.getInstrumentation().getAllLoadedClasses();
+        String rootFold = rootFile.substring(0, rootFile.lastIndexOf("/"));
```

Comment:
Seems awkward to have to construct this `rootFold` based off of `rootFile`. What is `rootFile` used for ultimately? If there is not that much use for needing a directory to write things in (i.e., just a few files are needed for input/output), could those just be passed in as file names directly?